baseapp/baseapp.go

```diff
@@ -666,7 +666,8 @@ func (app *BaseApp) getContextForTx(mode execMode, txBytes []byte) sdk.Context {
 		panic(fmt.Sprintf("state is nil for mode %v", mode))
 	}
 	ctx := modeState.Context().
-		WithTxBytes(txBytes)
+		WithTxBytes(txBytes).
+		WithGasMeter(storetypes.NewInfiniteGasMeter())
```

Comment:
I see you've added the backport tag, but is this really backportable?

Reply:
I guess so, the gas meter will be replaced by ante handler, so whatever recorded in the initial gas meter is discarded there.

Reply:
The root gas meter is an infinite gas meter. This change solves the race as the root gas meter is not shared anymore with the new context.  Looks good for me